index.js

```diff
@@ -54,14 +54,48 @@ module.exports = {
 
           const navigate = Page.enable()
             .then(() => Page.navigate({ url }))
+            .then(() => Page.addScriptToEvaluateOnNewDocument({source: `
+              window.addEventListener('application', ({ detail: Application }) => {
+                Application.autoboot = false;
+              });
+            `}))
             .then(() => Page.loadEventFired());
 
           return navigate
             .then(() => Runtime.evaluate({ awaitPromise: true, expression: `
-              ${this._appGlobal()}.visit('${visitPath}')
-                .then((application) => {
-                  return document.body.querySelector('.ember-view').outerHTML;
-                });
+              // Wrap inside a native promise since appGlobal.visit returns
+              // RSVP.Promise which doesn't play well with awaitPromise.
```

Comment:
I wonder if this is an issue with the JS lib and not headless. I am using the result of `RSVP.Promise` in an Elixir app that uses headless and it works just fine. I'll dig into this a bit.

Reply:
Thanks - I couldn't actually figure out the root cause and it's driving me nuts. I just know that I works with the Promise wrap.